src/diagram/components/diagram.tsx

```diff
@@ -28,14 +28,20 @@ const edgeTypes = {
 
 interface IProps {
   dqRoot: DQRootType;
+  showEditVariableDialog?: () => void;
   showNestedSet?: boolean;
   getDiagramExport?: () => unknown;
 }
-export const _Diagram = ({ dqRoot, getDiagramExport }: IProps) => {
+export const _Diagram = ({ dqRoot, getDiagramExport, showEditVariableDialog }: IProps) => {
   const reactFlowWrapper = useRef<any>(null);
-  const [ ,setSelectedNode] = useState<DQNodeType | undefined>();
+  const [selectedNode, setSelectedNode] = useState<DQNodeType | undefined>();
   const [rfInstance, setRfInstance] = useState<any>();
 
+  // Keep the model's selected node up to date with ReactFlow's
+  useEffect(() => {
+    dqRoot.setSelectedNode(selectedNode);
+  }, [dqRoot, selectedNode]);
+
```

Comment:
Rather than this approach, I'd suggest just calling `dqRoot.setSelectedNode(selectedNode);` in the onSelectionChange handler.  This will make the selection update faster instead of having to wait for a render loop to happen.
Then I think you can get rid of the setSelectedNode state all together. 

I remember now setSelectedNode was here. We used to have a form (like the one you are adding here). And it was rendered based on the selectedNode state.